utils/record/record.go

```diff
@@ -207,12 +207,21 @@ func (e *EventRecorderAdapter) defaultEventf(object runtime.Object, warn bool, o
 		if kind == "Rollout" {
 			e.RolloutEventCounter.WithLabelValues(namespace, name, opts.EventType, opts.EventReason).Inc()
 		}
-		err := e.sendNotifications(object, opts)
+
+		apis, err := e.apiFactory.GetAPIsFromNamespace(namespace)
 		if err != nil {
-			logCtx.Errorf("Notifications failed to send for eventReason %s with error: %s", opts.EventReason, err)
+			logCtx.Errorf("notifications failed to get apis for eventReason %s with error: %s", opts.EventReason, err)
 			e.NotificationFailedCounter.WithLabelValues(namespace, name, opts.EventType, opts.EventReason).Inc()
```

Comment:
Wonder if we want to make this also driven by the flag of self service enabled?

Reply:
I had to think about this but I like keeping this code the same, it's just one less branch in logic. This works because the informers we pass into notification engine will not be watching the other namespaces.